tkg/client/upgrade_region.go

```diff
@@ -260,6 +260,20 @@ func (c *TkgClient) validateAndconfigure(options *UpgradeClusterOptions, regiona
 		if err != nil {
 			return errors.Wrap(err, "error getting ami id")
 		}
+	} else if providerType == VSphereProviderName {
+		vcClient, err := c.GetVSphereEndpoint(regionalClusterClient)
+		if err != nil {
+			return NewValidationError(ValidationErrorCode, errors.Wrap(err, "failed to get VC client").Error())
+		}
+
+		dc, err := c.TKGConfigReaderWriter().Get(constants.ConfigVariableVsphereDatacenter)
+		if err != nil {
+			return NewValidationError(ValidationErrorCode, errors.Errorf("failed to get %s", constants.ConfigVariableVsphereDatacenter).Error())
+		}
+
+		if err := c.ValidateVsphereResources(vcClient, dc); err != nil {
```

Comment:
We need to figure out when and how the full path values are changed in the secret tkg-pkg-tkg-system-values. The code logical is suppose to reuse the same value with full path.